File: server/entities/content-page.js
const Realm = require('realm');
const { ContentElement } = require('./content-element');

function ContentPage() { }

ContentPage.schema = {
  name: 'ContentPage',
  primaryKey: 'uuid',
  properties: {
    title: {type: 'string', optional: true},
    elements: {type: 'list', objectType: 'ContentElement'},
    priority: {type: 'int', default: 0, indexed: true},
    mainColor: {type: 'string', optional: true},
    lang: {type: 'string', optional: true},
    tag: {type: 'string', indexed: true},
    uuid: {type: 'string'}
  }
};

exports.ContentPage = ContentPage;

File: android/app/src/main/java/io/realm/conference/data/entity/ContentPage.kt
package io.realm.conference.data.entity

import io.realm.RealmList
import io.realm.RealmModel
import io.realm.annotations.Index
import io.realm.annotations.PrimaryKey
import io.realm.annotations.RealmClass

@RealmClass
open class ContentPage : RealmModel {

    @PrimaryKey
    var uuid: String = ""

    @Index
    var tag: String = ""

    @Index
    var priority: Long = 0
    var mainColor: String? = null
    var lang: String? = null
    var title: String? = null
    var elements: RealmList<ContentElement>? = null

}
